internal/cloudapi/server.go

```diff
@@ -204,9 +204,48 @@ func (server *Server) Compose(w http.ResponseWriter, r *http.Request) {
 				t.ImageName = key
 			}
 
+			targets = append(targets, t)
+		} else if uploadRequest.Type == "gcp" {
+			var gcpUploadOptions GCPUploadRequestOptions
+			jsonUploadOptions, err := json.Marshal(uploadRequest.Options)
+			if err != nil {
+				http.Error(w, "Unable to marshal gcp upload request", http.StatusInternalServerError)
+				return
+			}
+			err = json.Unmarshal(jsonUploadOptions, &gcpUploadOptions)
+			if err != nil {
+				http.Error(w, "Unable to unmarshal gcp upload request", http.StatusInternalServerError)
+				return
+			}
+
+			var share []string
+			if gcpUploadOptions.ShareWithAccounts != nil {
+				share = *gcpUploadOptions.ShareWithAccounts
+			}
+			var region string
+			if gcpUploadOptions.Region != nil {
+				region = *gcpUploadOptions.Region
+			}
+			object := fmt.Sprintf("composer-api-%s", uuid.New().String())
+			t := target.NewGCPTarget(&target.GCPTargetOptions{
+				Filename:          imageType.Filename(),
+				Region:            region,
+				Os:                "", // not exposed in cloudapi for now
+				Bucket:            gcpUploadOptions.Bucket,
+				Object:            object,
+				ShareWithAccounts: share,
+			})
+			// TODO: randomize the image name by appending some suffix?
```

Comment:
I would say: either let the caller decide the name, and require them to choose a unique one, or don't and use the random string as below.

Reply:
The problem from my point of view is that the image name must be unique within the GCP Project that we will use on our side for importing and sharing the image. Since the user won't see what image names we already have within the project, they could easily choose an image name that already exists. The unfortunate thing is that the workflow will fail quite late while importing the already uploaded image. So there is currently no real feedback when creating the compose request if the chosen image name already exists.

Nevertheless the internal GCP API could be extended with a function which would check if an image with the given name already exists within the GCP Project. But there could be race conditions (especially when more users use the API simultaneously) and also in order to expose this in cloudapi, the check would probably need to be done by a worker, which has GCP credentials.

So since I don't see any ideal solution here, I'll incline to whatever you think will be OK for the user...

Reply:
We could leave this up to the caller (`image-builder` and make them responsible for avoiding clashes), but they'd basically be left with the same problem of having to pick a random name, they couldn't actually expose it to the user for the reasons you point out. I'd be ok with keeping it exactly as you have it here, then either `image-builder` would provide a name (appropriately randomized) or we'd create a random one internally.

Reply:
@Gundersanne do you have any preference here? I basically did it the same way as AWS part does it. If the image name (snapshot name for AWS) was provided by the caller, just use it, otherwise use `composer-api-<UUID>`...

Reply:
`fmt.Sprintf("composer-api-%s", uuid.New().String())` is fine for now I think. It's useful to have the default be some unique thing which is useful during testing.

Reply:
OK, thanks for all the comments. I'll just remove the "TODO" comment.